code/src/OSFramework/Grid/AbstractGrid.ts

```diff
@@ -249,7 +249,9 @@ namespace OSFramework.Grid {
                     this.dataSource.getMetadata(),
                     this.config.allowEdit
                 );
-                generated.forEach((p) => this.addColumn(p));
+                if (this._columns.size === 0) {
+                    generated.forEach((p) => this.addColumn(p));
+                }
```

Comment:
Does this cover the scenario, when a new column is added as well?

Reply:
I was checking the same and this code only runs in auto generated columns that have metadata (where created with ArrangeData), so no problem with add/remove columns

Reply:
As discussed, we will leave for now as it is. @glundgren93 if you can add just an improvement to backlog, to (maybe) support swapping data sources in the future.

Reply:
yes!